crates/starknet-server/src/api/http/endpoints/postman.rs

```diff
@@ -1,30 +1,158 @@
+    // will create L2 to L1 messages.
+    let mut starknet = state.api.starknet.write().await;
+
+    let dry_run = data.dry_run.unwrap_or(false);
+
+    let messages_to_l2 = if dry_run {
+        Ok(vec![])
+    } else {
+        starknet
+            .fetch_and_execute_messages_to_l2()
+            .await
+            .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?
+            .into_iter()
+            .map(|m| m.try_into())
+            .collect::<Result<Vec<MessageToL2>, HttpApiError>>()
+    };
+
+    let from_block = if let Some(m) = &starknet.messaging { m.last_local_block } else { 0 };
+
+    let (messages_to_l1, last_local_block) = if dry_run {
+        (
+            starknet
+                .collect_messages_to_l1(from_block)
+                .await
+                .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?
+                .into_iter()
+                .map(|m| m.try_into())
+                .collect::<Result<Vec<MessageToL1>, HttpApiError>>(),
+            0,
+        )
+    } else {
+        let (msgs, b) = starknet
+            .collect_and_send_messages_to_l1(from_block)
+            .await
+            .map_err(|e| HttpApiError::MessagingError { msg: e.to_string() })?;
+
+        (
+            msgs.into_iter()
+                .map(|m| m.try_into())
+                .collect::<Result<Vec<MessageToL1>, HttpApiError>>(),
+            b,
+        )
+    };
+
+    let l1_provider = if dry_run {
+        "dry run".to_string()
+    } else {
+        starknet.messaging_url().unwrap_or("Not set".to_string())
+    };
+
+    if !dry_run {
+        // +1 to ensure this last block is not collected anymore.
+        starknet.messaging.as_mut().expect("Messaging expected configured").last_local_block =
+            last_local_block + 1;
+    }
+
+    match (messages_to_l1, messages_to_l2) {
+        (Ok(l1s), Ok(l2s)) => {
+            Ok(Json(FlushedMessages { messages_to_l1: l1s, messages_to_l2: l2s, l1_provider }))
+        }
+        (Ok(l1s), Err(e)) => {
+            if dry_run {
+                Ok(Json(FlushedMessages {
+                    messages_to_l1: l1s,
+                    messages_to_l2: vec![],
+                    l1_provider,
+                }))
+            } else {
+                Err(HttpApiError::MessagingError { msg: format!("MessagesToL2: {}", e) })
+            }
+        }
+        (Err(e), Ok(l2s)) => {
+            if dry_run {
+                Ok(Json(FlushedMessages {
+                    messages_to_l1: vec![],
+                    messages_to_l2: l2s,
+                    l1_provider,
+                }))
+            } else {
+                Err(HttpApiError::MessagingError { msg: format!("MessagesToL1: {}", e) })
+            }
+        }
+        (Err(e1), Err(e2)) => Err(HttpApiError::MessagingError {
+            msg: format!("MessagesToL1: {} || MessagesToL2: {}", e1, e2),
+        }),
+    }
```

Comment:
I notice flush is now returning Error if it cannot flush, whereas devnet-py simply returned `{"generated_l2_transactions":[]}`. In fact, the whole response of `/postman/flush` seems to be changed. Do you think devnet-py had it wrong?

Reply:
I actually took the output from the video you sent to me on Discord. And there were no `generated_l2_transactions`. But it's a very good point so I added it into the flush result.

```rust
#[derive(Serialize)]
pub(crate) struct FlushedMessages {
    pub messages_to_l1: Vec<MessageToL1>,
    pub messages_to_l2: Vec<MessageToL2>,
    pub generated_l2_transactions: Vec<TransactionHash>,
    pub l1_provider: String,
}
```
I do think that only having the transaction hashes is too low for a devX perspective when debugging messaging. But having those hashes makes a lot's of sense.

Also, now if there is an error, I don't think we should return empty arrays. Having the corresponding error is required as the user will know exactly where the error happened and he will be able to solve it (or for us very helpful to debug ^^).

Reply:
Sure if there is an error we shold inform the user of the error. But my real question is, is it really an error if the users wishes to flush when there is nothing to flush? Seems like we used to treat it as an idempotent operation. That's what I asked if you consider wrong.

Reply:
Sorry didn't get you right.
No, it's not an error to flush if there is no messages. And now the behavior is modified to return empty arrays if the flush endpoint is queried and no message is available. :+1: